MQ2FeedMe.cpp

```diff
@@ -126,239 +227,620 @@ void ListTypes(const std::list<std::string>& list)
+		}
+	}
+	else if (ci_equals(Arg, "warn"))
+	{
+		if (bDrinkWarn)
+		{
+			bDrinkWarn = false;
 			WriteChatf("\ay%s\aw:: Drink Warning Off", PLUGIN_NAME);
-		} else {
-			bDrinkWarn = 1;
+		}
+		else {
+			bDrinkWarn = true;
 			WriteChatf("\ay%s\aw:: Drink Warning On", PLUGIN_NAME);
 		}
 	}
-	else if (!_stricmp(Arg, "reload")) {
-		ReadList(&Hunger, "FOOD");
-		ReadList(&Thirst, "DRINK");
+	else if (ci_equals(Arg, "add"))
+	{
+		HandleAddFoodDrinkItem();
 	}
-	else if (!_stricmp(Arg, "add")) {
-		if (!ItemOnCursor()) {
-			WriteChatf("%s:: \arNeed to have a food item on your cursor to do this.", PLUGIN_NAME);
+	else if (ci_equals(Arg, "remove"))
+	{
+		char item[MAX_STRING] = { 0 };
+		strcpy_s(item, GetNextArg(szLine));
+		if (item[0] == '\0')
+		{
 			return;
 		}
-
-		ItemPtr pItem = GetPcProfile()->GetInventorySlot(InvSlot_Cursor);
-		if (pItem) {
-			if (!pItem->GetItemDefinition()->FoodDuration || pItem->GetItemClass() != ItemClass_Drink) {
-				WriteChatf("%s:: \arThat's not a drink. Don't be ridiculous", PLUGIN_NAME);
-				return;
-			}
-			WriteChatf("%s:: \ayFound Item: \ap%s", PLUGIN_NAME, pItem->GetName());
-			int FoodIndex = (int)Thirst.size() + 1;
-			for (const std::string& itemName : Thirst) {
-				if (ci_equals(itemName, pItem->GetName())) {
-					WriteChatf("%s:: \ap%s \aris already on the list", PLUGIN_NAME, pItem->GetName());
-					return;
-				}
-			}
-			WritePrivateProfileString("Drink", fmt::format("Drink{}", FoodIndex), pItem->GetName(), INIFileName);
-			Thirst.push_back(pItem->GetName());
-			EzCommand("/autoinv");
-			WriteChatf("%s \agAdded\aw: \ap%s \ayto your autodrink list", PLUGIN_NAME, pItem->GetName());
-		}
-	}
-	else if (IsNumber(Arg)) {
-		DrnkAt = atoi(Arg);
-		if (DrnkAt < 0) DrnkAt = 0;
-		else if (DrnkAt > 5000) DrnkAt = 5000;
-		sprintf_s(Buffer, "%d", DrnkAt);
-		WritePrivateProfileString(GetCharInfo()->Name, "AutoDrink", Buffer, INIFileName);
-		WriteChatf("\ay%s\aw:: AutoDrink(\ag%s\ax).", PLUGIN_NAME, (DrnkAt) ? Buffer :"\aroff");
-		if (bAnnLevels) WriteChatf("\ay%s\aw:: Current Thirst(\ag%d\ax) Hunger(\ag%d\ax)", PLUGIN_NAME, GetPcProfile()->thirstlevel, GetPcProfile()->hungerlevel);
-	}
-	else if (!_stricmp(Arg, "announceConsume")) {
-		if (bAnnConsume) {
-			bAnnConsume = false;
-			WriteChatf("\ay%s\aw::Consumption Notification Off", PLUGIN_NAME);
+		// check both vectors for the item
+		// and remove
+		HandleRemoveFoodDrinkItem("drink", item);
+	}
+	else if (IsNumber(Arg))
+	{
+		iDrinkAt = GetIntFromString(Arg, 5000);
+		if (iDrinkAt < 0)
+		{
+			iDrinkAt = 0;
 		}
-		else {
-			bAnnConsume = true;
-			WriteChatf("\ay%s\aw::Consumption Notification On", PLUGIN_NAME);
+		else if (iDrinkAt > 5000)
+		{
+			iDrinkAt = 5000;
+		}
+
+		WritePrivateProfileInt(GetCharInfo()->Name, "AutoDrink", iDrinkAt, INIFileName);
+		WriteChatf("\ay%s\aw:: AutoDrink (\ag%s\ax).", PLUGIN_NAME, iDrinkAt ? std::to_string(iDrinkAt).c_str() : "\aroff");
```

Comment:
The way this is overloaded is unintuitive and not great. Only do it with thte parameter types are the same on both sides of the condition and you don't have to jump through hoops to make it work.

```suggestion
		if (iDrinkAt)
		{
			WriteChatf("\ay%s\aw:: AutoDrink (\ag%d\ax).", PLUGIN_NAME, iDrinkAt);
		}
		else
		{
			WriteChatf("\ay%s\aw:: AutoDrink (\aroff\ax).", PLUGIN_NAME);
		}
```